components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -229,13 +230,15 @@ impl CanvasRenderingContext2D {
             HTMLImageElementOrHTMLCanvasElementOrCanvasRenderingContext2D::CanvasRenderingContext2D(image) =>
                 image.origin_is_clean(),
             HTMLImageElementOrHTMLCanvasElementOrCanvasRenderingContext2D::HTMLImageElement(image) =>
-                match image.get_url() {
-                    None => true,
-                    Some(url) => {
-                        // TODO(zbarsky): we should check the origin of the image against
-                        // the entry settings object, but for now check it against the canvas' doc.
-                        let node: &Node = &*self.canvas.upcast();
-                        url.origin() == node.owner_doc().url().origin()
+                match image.get_origin() {
```

Comment:
We should be able to use `image.get_origin.expect("Image's origin is missing")` here, since `is_origin_clean` should only be called after we have a valid response for the image.